GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -52,7 +52,7 @@ public void CountLooseObjects(out int count, out long size)
 
                 if (GitObjects.IsLooseObjectsDirectory(directoryName))
                 {
-                    string dirPath = Path.Combine(this.Context.Enlistment.GitObjectsRoot, directoryPath);
+                    string dirPath = Path.Combine(this.Context.Enlistment.GitObjectsRoot, directoryName);

```

Comment:
We could use `directoryPath` and delete this line right?

Reply:
@kewillford, @derrickstolee another change I've been considering is replacing this line:

`string directoryName = directoryPath.TrimEnd(Path.DirectorySeparatorChar).Split(Path.DirectorySeparatorChar).Last();`

With

`Path.GetDirectoryName(directoryPath)`

Thoughts?  I was a bit on the fence about how much cleanup to do here.

Reply:
That looks much simpler. This is likely the only time we have to do that cleanup, so please do!

Reply:
Yeah I thought about commenting on that as well and since we were both thinking it then it is probably a good thing to change.

Reply:
> Yeah I thought about commenting on that as well and since we were both thinking it then it is probably a good thing to change.

I am actually going to back the change to use `Path.GetDirectoryName(directoryPath)` out for now.  It ends up that `Path.GetDirectoryName(...)` doesn't play well with the `mock:` paths we use in the unit tests.

I've filed #1572 to clean this up.